File: py-sdk/sdk/sdk/path_utils/path_utils.py
import os
from abc import ABC, abstractmethod
from dataclasses import dataclass

from vyper import v


@dataclass
class PathUtilsABC(ABC):
    @staticmethod
    @abstractmethod
    def get_base_path() -> str:
        pass

    @staticmethod
    @abstractmethod
    def compose_path(*relative_path: str) -> str:
        pass


@dataclass
class PathUtils(PathUtilsABC):
    @staticmethod
    def get_base_path() -> str:
        return v.get_string("metadata.base_path")

    @staticmethod
    def compose_path(*relative_path: str) -> str:
        base_path = PathUtils.get_base_path()
        return os.path.join(base_path, *relative_path) if relative_path else base_path

File: go-sdk/sdk/path-utils/path_utils.go
package pathutils

import (
	path2 "path"

	"github.com/konstellation-io/kai-sdk/go-sdk/internal/common"

	"github.com/go-logr/logr"
	"github.com/spf13/viper"
)

type PathUtils struct {
	logger logr.Logger
}

func NewPathUtils(logger logr.Logger) *PathUtils {
	return &PathUtils{
		logger: logger.WithName("[PATH UTILS]"),
	}
}

func (pu PathUtils) GetBasePath() string {
	return viper.GetString(common.ConfigMetadataBasePathKey)
}

func (pu PathUtils) ComposePath(relativePath ...string) string {
	path := pu.GetBasePath()
	if len(relativePath) == 0 {
		return path
	}

	for _, elem := range relativePath {
		path = path2.Join(path, elem)
	}

	return path
}
